desc/objectives/objective_funs.py

```diff
@@ -809,81 +774,16 @@ def __init__(
-        # if args is already set don't overwrite it
-        self._args = getattr(
-            self,
-            "_args",
-            [arg for arg in getfullargspec(self.compute)[0] if arg != "self"],
-        )
-        self._eq = eq
-        if eq is None:
-            warnings.warn(
-                FutureWarning(
-                    "Creating an Objective without specifying the Equilibrium to"
-                    " optimize is deprecated, in the future this will raise an error."
-                )
-            )
-
-    def _set_dimensions(self, eq):
-        """Set state vector component dimensions."""
-        self._dimensions = {}
-        for arg in arg_order:
-            self._dimensions[arg] = np.atleast_1d(getattr(eq, arg)).size
+        self._things = flatten_list([things], True)
 
     def _set_derivatives(self):
         """Set up derivatives of the objective wrt each argument."""
-        self._derivatives = {
-            "jac_scaled": {},
-            "jac_unscaled": {},
-            "grad": {},
-            "hess": {},
-        }
-
-        for arg in arg_order:
-            if arg in self.args:  # derivative wrt arg
-                self._derivatives["jac_unscaled"][arg] = Derivative(
-                    self.compute_unscaled,
-                    argnum=self.args.index(arg),
-                    mode="fwd",
-                )
-                self._derivatives["jac_scaled"][arg] = Derivative(
-                    self.compute_scaled,
-                    argnum=self.args.index(arg),
-                    mode="fwd",
-                )
-                self._derivatives["grad"][arg] = Derivative(
-                    self.compute_scalar,
-                    argnum=self.args.index(arg),
-                    mode="grad",
-                )
-                self._derivatives["hess"][arg] = Derivative(
-                    self.compute_scalar,
-                    argnum=self.args.index(arg),
-                    mode="hess",
-                )
-            else:  # these derivatives are always zero
-                self._derivatives["jac_unscaled"][
-                    arg
-                ] = lambda *args, arg=arg, **kwargs: jnp.zeros(
-                    (self.dim_f, self.dimensions[arg])
-                )
-                self._derivatives["jac_scaled"][
-                    arg
-                ] = lambda *args, arg=arg, **kwargs: jnp.zeros(
-                    (self.dim_f, self.dimensions[arg])
-                )
-                self._derivatives["grad"][
-                    arg
-                ] = lambda *args, arg=arg, **kwargs: jnp.zeros(
-                    (1, self.dimensions[arg])
-                )
-                self._derivatives["hess"][
-                    arg
-                ] = lambda *args, arg=arg, **kwargs: jnp.zeros(
-                    (self.dimensions[arg], self.dimensions[arg])
-                )
+        self._grad = Derivative(self.compute_scalar, mode="grad")
```

Comment:
we can do this without specifying arg indices now because all of the compute functiosn take in `params`?

Reply:
Exactly. Though for objectives that connect multiple objects we might need to modify this. These are currently unused so not an issue for now, but I'll fix it when working on #666

Reply:
objectives like `PlasmaVesselDistance`? or others

Reply:
For now yes thats the only one that connects two objectives